diff --git a/core/Makefile b/core/Makefile
index d3a2402..ef1365e 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -470,7 +470,7 @@ $(INSTALLED_BOOTIMAGE_TARGET): $(MKEXT2IMG) $(INTERNAL_BOOTIMAGE_FILES)
 	$(hide) $(MKEXT2BOOTIMG) $(INTERNAL_BOOTIMAGE_ARGS) --output $@
 	@echo -e ${CL_CYN}"Made boot image for Ubuntu Touch: $@"${CL_RST}
 
-else ifndef BOARD_CUSTOM_BOOTIMG_MK # TARGET_BOOTIMAGE_USE_EXT2 != true
+else #ifndef BOARD_CUSTOM_BOOTIMG_MK # TARGET_BOOTIMAGE_USE_EXT2 != true
 INTERNAL_BOOTIMAGE_ARGS += $(INTERNAL_BOOTIMAGE_ARGS_COMMON)
 INTERNAL_BOOTIMAGE_ARGS_ANDROID += $(INTERNAL_BOOTIMAGE_ARGS_COMMON)
 
diff --git a/core/main.mk b/core/main.mk
index 85bc117..f5fa86f 100644
--- a/core/main.mk
+++ b/core/main.mk
@@ -471,6 +471,7 @@ subdirs := \
 	build/tools/zipalign \
 	development/tools/emulator/opengl \
 	external/aac \
+	external/bluetooth/bluedroid \
 	external/busybox \
 	external/bzip2 \
 	external/e2fsprogs \
@@ -501,6 +502,7 @@ subdirs := \
 	external/webrtc \
 	external/wpa_supplicant_6 \
 	external/wpa_supplicant_8 \
+	external/wpa_supplicant_8_ti \
 	external/yaffs2 \
 	external/zlib \
 	frameworks/av/camera \
@@ -532,6 +534,7 @@ subdirs := \
 	frameworks/native/services \
 	frameworks/opt/emoji \
 	hardware \
+	libnativehelper \
 	prebuilts/tools/linux-x86/sdl \
 	sdk/emulator \
 	system/core \
