diff --git a/rootdir/init.rc b/rootdir/init.rc
index cfa459e..7beb5ab 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -147,7 +147,7 @@ on fs
     # Mount /system rw first to give the filesystem a chance to save a checkpoint
     mount yaffs2 mtd@system /system
     mount yaffs2 mtd@system /system ro remount
-    mount yaffs2 mtd@userdata /data nodev
+    mount ext4 /dev/block/mmcblk0p2 /data nodev
     mount yaffs2 mtd@cache /cache nosuid nodev
 
 on post-fs
